### PR TITLE
Add R-FCN incorporation and fine-tuning to 'test' brach.

### PR DESCRIPTION
**Pull request by Awcrr**
Add R-FCN incorporation and fine-tuning to 'test' brach.

### CODE REVIEW

File: models/preresnet.lua [hunk 1]
```diff
@@ -125,6 +125,42 @@ local function createModel(opt)
       return s
    end
 
+   -- RCN-Regional Convolutional Network
+   local function rcn(nInputPlane, nOutputPlane, multiFactor, w, h)
+      local s = nn.Sequential()
+      local nMiddle = nInputPlane / 2
+      s:add(Convolution(nInputPlane, nMiddle, 1, 1))
+      s:add(ReLU(true))
```
Comment (Cysu) [inline models/preresnet.lua]: May consider add a BatchNorm layer before ReLU.

File: models/rfcn.lua [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      local sOutside = nn.Sequential()
+      s:add(SBatchNorm(nInputPlane))
+      s:add(ReLU(true))
+      -- reduce dimension
+      local nMiddle = nInputPlane / 2 
+      local conv0 = Convolution(nInputPlane, nMiddle, 1, 1)
+      -- initialize
+      conv0.weight:normal(0, 0.01)
+      conv0.bias:zero()
+      s:add(conv0)
```
Comment (Cysu) [inline models/rfcn.lua:28]: May consider add a BatchNorm before this ReLU.

File: models/rfcn.lua [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            local offset4 = math.floor(j * h / multiFactor) + 1
+            if offset4 + len4 > h + 1 then
+               offset4 = w - len4 + 1
+            end
+            part:add(nn.Narrow(3, offset3, len3))
+            part:add(nn.Narrow(4, offset4, len4))
+            -- 1x1 Convolution
+            local conv1 = Convolution(nMiddle, nOutputPlane, 1, 1)
+            -- initialize
+            conv1.weight:normal(0, 0.01)
```
Comment (Cysu) [inline models/rfcn.lua]: Maybe we can tweak this initialization std, say, to 0.001.

File: models/rfcn.lua [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            part:add(conv1)
+            part:add(Max(len3, len4, 1, 1))
+            table:add(part)
+         end
+      end
+
+      s:add(table)
+      s:add(nn.CAddTable())
+      -- Balance the RFCN output and fc-layer output
+      local mul = nn.Mul()
```
Comment (Cysu) [inline models/rfcn.lua]: There seems to be an nn.MulConstant module for this purpose. But it's fine to keep the current implementation.

File: scripts/strategy.lua [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+for i = 1, #scales do
+   print("- Processing scale-" .. scales[i])
+   local scores = torch.load("scores_test/scores_" .. scales[i] .. ".t7") 
+   local tmpoutput = torch.Tensor(#scores, 1000):zero()
+   if not output then
+      output = torch.Tensor(#scores, 1000):zero()
+   end
+   if strategy == 1 then
+      for j = 1, #scores do 
+         local pooling = nn.SpatialAveragePooling(scores[j]:size(4), scores[j]:size(3), 1, 1)
```
Comment (Cysu) [inline scripts/strategy.lua]: Maybe we can just simply use tensor operations to compute the mean here instead of creating an nn layer each time. For example, `scores[j] = scores[j]:mean(3):mean(4):squeeze()` can compute the score vector. Might slightly accelerate the speed.

Comment (Awcrr) [reply scripts/strategy.lua]: Make sense :). I think it'll save some overhead time caused by 'nn'.
